src/nginx_config.py

```diff
@@ -252,7 +241,6 @@ def _build_server_config(
                     {"directive": "ssl_certificate_key", "args": [KEY_PATH]},
                     {"directive": "ssl_protocols", "args": ["TLSv1", "TLSv1.1", "TLSv1.2"]},
                     {"directive": "ssl_ciphers", "args": ["HIGH:!aNULL:!MD5"]},  # codespell:ignore
-                    *self._resolver(custom_resolver=self.dns_IP_address),
```

Comment:
Not all the uses of  `_resolver` were related to the pod restart. I wonder if we don't break anything with removing it from lines with `custom_resolver` specified.

Reply:
The resolver was pointing to kube-dns. We shouldn't need that since we'd be always deploying this inside a pod in the k8s cluster and we shouldn't have to manually specify a resolver by nature.

+ itests pass :D